### PR TITLE
Add colormap option to set_data + update jet implementation.

### PR DESCRIPTION
**Pull request by jdumas**
#### Check all that apply (change to `[x]`)
- [x] All changes meet [libigl style-guidelines](https://libigl.github.io/style-guidelines/).
- [x] This is a minor change.

### CODE REVIEW

**Comment by jdumas** [comment]
I'm also tempted to simply mark jet as deprecated and add the *turbo* colormap:
https://ai.googleblog.com/2019/08/turbo-improved-rainbow-colormap-for.html

**Comment by alecjacobson** [comment]
Yeah. I like this idea, too.

File: include/igl/opengl/ViewerData.cpp [hunk 1]
```diff
@@ -265,24 +263,25 @@ IGL_INLINE void igl::opengl::ViewerData::set_texture(
...
-    igl::parula(Eigen::VectorXd::LinSpaced(21,0,1).eval(),false,CM);
+    igl::colormap(cmap,Eigen::VectorXd::LinSpaced(256,0,1).eval(),0,1,CM);
     set_colormap(CM);
-  } 
+  }
   set_uv(((D.array()-caxis_min)/(caxis_max-caxis_min)).replicate(1,2));
 }
 
-IGL_INLINE void igl::opengl::ViewerData::set_data( const Eigen::VectorXd & D)
+IGL_INLINE void igl::opengl::ViewerData::set_data( const Eigen::VectorXd & D, igl::ColorMapType cmap)
```
Comment (alecjacobson) [inline include/igl/opengl/ViewerData.cpp]: Let's keep a `set_data(D)` that uses a default colormap. 

We should also use `const` for input params.

Comment (alecjacobson) [reply include/igl/opengl/ViewerData.cpp]: nvm about default. Found the default it in the .h

**Comment by jdumas** [comment]
Ok I switched jet to turbo. Here is a comparison:

Jet:
2019-12-01-165823_1280x800_scrot

Turbo:
2019-12-01-165843_1273x814_scrot

File: include/igl/opengl/ViewerData.cpp [hunk 2]
```diff
@@ -265,24 +263,25 @@ IGL_INLINE void igl::opengl::ViewerData::set_texture(
...
+  const Eigen::VectorXd & D,
+  double caxis_min,
+  double caxis_max,
+  igl::ColorMapType cmap)
 {
   if(!show_texture)
   {
     Eigen::MatrixXd CM;
-    igl::parula(Eigen::VectorXd::LinSpaced(21,0,1).eval(),false,CM);
+    igl::colormap(cmap,Eigen::VectorXd::LinSpaced(256,0,1).eval(),0,1,CM);
```
Comment (alecjacobson) [inline include/igl/opengl/ViewerData.cpp]: this `256` should be reverted `21`.

Intervals are (by default) better for visualizing scalar fields than an effectively continuous map.

Comment (jdumas) [reply include/igl/opengl/ViewerData.cpp]: Ok I see. I've added an argument to set the number of steps/interval then, with default = 21.

**Comment by jdumas** [comment]
Btw I'm also questioning if we should make Parula our default colormap. Matplotlib developed Viridis and co. because Parula is copyrighted by Matworks. Do we really want to make it our default colormap, or should we stick with Viridis?

**Comment by alecjacobson** [comment]
that's fine with me.

>

**Comment by jdumas** [comment]
That's fine meaning ok to switch to viridis, or ok to keep parula?

**Comment by alecjacobson** [comment]
sorry I was unclear. Switching to Viridis is fine. As long as it's close
enough looking to Parula that I don't need to remake all of the new
tutorial figures...

>

**Comment by alecjacobson** [comment]
fwiw, I much more impressed with the colorbrewer colormaps than matlab's or
matplotlib's. Maybe we should support those and use one as a default.

>

**Comment by jdumas** [comment]
You mean [this website](http://colorbrewer2.org)? Their colormaps are defined [here](https://github.com/axismaps/colorbrewer/blob/master/colorbrewer_schemes.js). Should be easy to integrate into libigl.

**Comment by jdumas** [comment]
I've switched to Viridis by default. Should be close enough to Parula to not perturb the tutorials. I'll merge once the build passes. We can add the colorbrew palette in a separate PR. Given that these are discrete colormaps, we can add a function to directly return the colorbar given the number of steps required by the user.

**Pull request merged**